jetty-client/src/test/java/org/eclipse/jetty/client/util/SPNEGOAuthenticationTest.java

```diff
@@ -0,0 +1,286 @@
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.client.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.kerby.kerberos.kerb.server.SimpleKdcServer;
+import org.eclipse.jetty.client.AbstractHttpClientServerTest;
+import org.eclipse.jetty.client.EmptyServerHandler;
+import org.eclipse.jetty.client.api.Authentication;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.security.ConstraintMapping;
+import org.eclipse.jetty.security.ConstraintSecurityHandler;
+import org.eclipse.jetty.security.HashLoginService;
+import org.eclipse.jetty.security.SpnegoLoginService2;
+import org.eclipse.jetty.security.authentication.AuthorizationService;
+import org.eclipse.jetty.security.authentication.SpnegoAuthenticator2;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.session.DefaultSessionIdManager;
+import org.eclipse.jetty.server.session.SessionHandler;
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.security.Constraint;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class SPNEGOAuthenticationTest extends AbstractHttpClientServerTest
+{
+    private static final Logger LOG = Log.getLogger(SPNEGOAuthenticationTest.class);
+
+    static
+    {
+        if (LOG.isDebugEnabled())
+        {
+            System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", "debug");
```

Comment:
If SimpleLogger was initialized before this test (in say another test) then this system property change will have no impact.

Reply:
It's the only way I could find to setup Kerby logging, which uses SLF4J.
What I do is to include in the POM a dependency for `slf4j-simple` to have Kerby logging.
How can I route SLF4J from Kerby to Jetty logging?

Reply:
Since its a test case, its not that important.
You'll want debugging when you are troubleshooting, etc.
Perhaps this should be changed away from "debug" to "info" before being merged back into mainline jetty tho?